[BUG] CEA-708 hangs forever if character set is specified on rust builds

Can be verified by running this [sample](https://sampleplatform.ccextractor.org/sample/b5d6aad89f768674175fedf577b6df66a22f572f732d8a760c8e57b5b504829e) using the following arguments: `--service 1[EUC-KR]`.

It works as expected on a build of ccextractor built `-without-rust` but hangs forever otherwise. This is currently causing the sample platform tests to abort from a timeout as evidenced by this [run](https://sampleplatform.ccextractor.org/test/5668) taking more than 3 hours:
```
[INFO] Starting with entry 10 of 14
[WARN] Aborting CCExtractor, maximum time elapsed.
[ERROR] Path is empty
[INFO] Finished entry 10 with exit code: -1
[INFO] Starting with entry 11 of 14
[WARN] Aborting CCExtractor, maximum time elapsed.
[ERROR] Path is empty
[INFO] Finished entry 11 with exit code: -1
[INFO] Starting with entry 12 of 14
[ERROR] Path is empty
[INFO] Finished entry 12 with exit code: 0
[INFO] Starting with entry 13 of 14
[WARN] Aborting CCExtractor, maximum time elapsed.
[ERROR] Path is empty
[INFO] Finished entry 13 with exit code: -1
[INFO] Starting with entry 14 of 14
[ERROR] Path is empty
[INFO] Finished entry 14 with exit code: 0
[INFO] Runtime: 03:21:00.9565500
```

--- a/src/rust/Cargo.toml
+++ b/src/rust/Cargo.toml
@@ -13,7 +13,6 @@ crate-type = ["staticlib"]
 [dependencies]
 log = "0.4.26"
 env_logger = "0.8.4"
-iconv = "0.1.1"
 palette = "0.6.1"
 rsmpeg = { version = "0.14.2", optional = true, features = [
     "link_system_ffmpeg",
@@ -28,6 +27,7 @@ cfg-if = "1.0.0"
 num-integer = "0.1.46"
 lib_ccxr = { path = "lib_ccxr" }
 url = "2.5.4"
+encoding_rs = "0.8.5"
 
 [build-dependencies]
 bindgen = "0.64.0"

--- a/src/rust/src/decoder/tv_screen.rs
+++ b/src/rust/src/decoder/tv_screen.rs
@@ -17,6 +17,7 @@ use crate::{
     bindings::*,
     utils::{is_false, is_true},
 };
+use encoding_rs::Encoding;
 
 use log::{debug, warn};
 
@@ -232,8 +233,17 @@ impl dtvcc_tv_screen {
                 };
                 debug!("Charset: {}", charset);
 
-                let op = iconv::decode(&buf, charset).map_err(|err| err.to_string())?;
-                writer.write_to_file(op.as_bytes())?;
+                // Look up the encoding by label (name)
+                if let Some(encoding) = Encoding::for_label(charset.as_bytes()) {
+                    let (cow, _encoding_used, had_errors) = encoding.decode(&buf);
+                    if had_errors {
+                        println!("Warning: Had errors during encoding from {charset} to UTF-8");
+                    }
+                    if _encoding_used != encoding {
+                        println!("Warning: Encoding specified ({}) does not match encoding detected ({})",charset,_encoding_used.name());
+                    }
+                    writer.write_to_file(cow.as_bytes())?;
+                }
             }
         } else {
             writer.write_to_file(&buf)?;

